lib/fusion_auth/plugs/authorize_jwt.ex

```diff
@@ -62,21 +63,34 @@ defmodule FusionAuth.Plugs.AuthorizeJWT do
          {:ok, claims} <- verify_signature(token),
          {:ok, diff} <- verify_exp(claims["exp"], generate_refresh_token) do
       conn =
-        Plug.Conn.register_before_send(conn, fn conn ->
-          with true <- needs_refresh?(diff, generate_refresh_token, options[:refresh_window]),
-               {:ok, refresh} <- Utils.fetch_refresh(conn),
-               {:ok, %{"token" => new_token}, _} =
-                 FusionAuth.JWT.refresh_jwt(client, refresh, token) do
-            Plug.Conn.put_resp_header(
-              conn,
-              Application.get_env(:fusion_auth, :token_header_key),
-              new_token
-            )
-          else
-            _ ->
+        with true <- needs_refresh?(diff, generate_refresh_token, options[:refresh_window]),
+             {_, refresh} <- Utils.fetch_refresh(conn),
+             {:ok, %{"token" => new_token}, _} <-
+               FusionAuth.JWT.refresh_jwt(client, refresh, token) do
+          Plug.Conn.put_resp_header(
+            conn,
+            Application.get_env(:fusion_auth, :token_header_key),
+            new_token
+          )
+        else
+          {:error, body, env} ->
```

Comment:
There are three function calls in the `with` statements, are we sure we handle all the error cases for each of these function calls in the `else` block?

Reply:
yeah, I wrote tests for each case

Reply:
should it log the error for those error cases?

Reply:
Actually maybe we shouldn't be logging here from this package. Should we be logging from the consuming application for this package?

Reply:
id generally opt for logging in the consuming package

Reply:
Yeah thinking that might be the better approach here

Reply:
hm okay, so how do we want to bubble up this info?

Reply:
it also looks like we already have fusionauth logging errors anytime a result status is >= 300